casper/src/test/scala/coop/rchain/casper/genesis/PosUpdateSpec.scala

```diff
@@ -0,0 +1,177 @@
+package coop.rchain.casper.genesis
+
+import cats.syntax.all._
+import coop.rchain.casper.genesis.contracts.ProofOfStake
+import coop.rchain.casper.helper.TestNode
+import coop.rchain.casper.helper.TestNode._
+import coop.rchain.casper.util.ConstructDeploy
+import coop.rchain.crypto.PrivateKey
+import coop.rchain.crypto.signatures.Secp256k1
+import coop.rchain.models.GDeployId
+import coop.rchain.p2p.EffectsTestInstances.LogicalTime
+import coop.rchain.shared.Base16
+import coop.rchain.shared.scalatestcontrib._
+import coop.rchain.models.syntax._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.build.CompiledRholangTemplate
+import coop.rchain.rholang.interpreter.RhoType.{Boolean, String, Tuple2}
+import coop.rchain.rholang.interpreter.util.RevAddress
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Inspectors, Matchers}
+
+class PosUpdateSpec extends FlatSpec with Matchers with Inspectors {
+
+  import coop.rchain.casper.util.GenesisBuilder._
+
+  implicit val timeEff           = new LogicalTime[Effect]
+  private val shardId            = "root"
+  private val hexP1              = "fc743bd08a822d544bfbe05a5663fc325039a44c8f0c7fbea95a85517da5c36b"
+  private val hexP2              = "6e88cf274735f3f7f73ec3d7f0362c439ab508427682b5bd788007aca665d810"
+  private val hexP3              = "87369d132ed6a7626dc4c5dfbaf41e954dd0ec55830e613a3f868c74d64a7a22"
+  private val p1                 = PrivateKey(hexP1.unsafeDecodeHex)
+  private val pub1               = Secp256k1.toPublic(p1)
+  private val noPermissionKey    = PrivateKey(hexP2.unsafeDecodeHex)
+  private val noPermissionKeyPub = Secp256k1.toPublic(noPermissionKey)
+  private val p3                 = PrivateKey(hexP3.unsafeDecodeHex)
+  private val pub3               = Secp256k1.toPublic(p3)
+  private val validatorKeys      = Seq((p1, pub1), (noPermissionKey, noPermissionKeyPub), (p3, pub3))
+  private val genesisVaults =
+    Seq((p1, 100000000000L), (noPermissionKey, 100000000000L), (p3, 100000000000L))
+  private val bonds       = Map(pub1 -> 1000000L, noPermissionKeyPub -> 1000000L, pub3 -> 1000000L)
+  private val genesis     = buildGenesis(buildGenesisParameters(validatorKeys, genesisVaults, bonds))
+  private val epochLength = 1
+
+  private val updatePosTerm = CompiledRholangTemplate.loadTemplate(
+    "UpdatePos/ProposePos.rho",
```

Comment:
Please create simplest possible example of updating PoS contract like updating minimum bonds. Update code should not repeat the whole PoS contract from scratch. Is this possible?

---

Please also rename ProposePos.rho to UpdatePos.rho to be consistent with other examples.

Reply:
> Please create simplest possible example of updating PoS contract like updating minimum bonds. 
That's how the codes work now.

> Update code should not repeat the whole PoS contract from scratch. Is this possible?
No.